src/mame/drivers/gi6809.cpp

```diff
@@ -0,0 +1,1039 @@
+					break;
+
+			case 1: m_stage++;
+					m_cast_ca1 = false;
+					m_cast_ca2 = true;
+					m_cast_sense = 0x7f;  // lower coin
+					break;
+
+			case 2: m_stage++;
+					m_cast_ca1 = true;
+					m_cast_ca2 = true;
+					m_cast_sense = 0xff;  // none
+					break;
+
+			case 3: m_stage = 0;
+					break;
+		}
+	}
+	m_vsync = state;
+}
+
+
+WRITE_LINE_MEMBER( gi6809_state::crtc_vs_je )
+{
+	m_vsync = state;
+	m_pia[1]->cb1_w(state);   // IRQB-> IRQ Master
+	m_pia[1]->ca1_w(!state);  // IRQA-> FIRQ Master
+	m_pia[0]->ca1_w(state);   // IRQA-> IRQ Slave
+}
+
+
+READ_LINE_MEMBER( gi6809_state::ret_vsync )
+{
+	return m_vsync;
+}
+
+
+READ_LINE_MEMBER( gi6809_state::pia0_ca1 )
+{
+	m_pia[1]->read(1);
+	return m_vsync;
+}
+
+
+READ_LINE_MEMBER( gi6809_state::cast_pia1_ca1 )  // castaway - coin system (A) - Hi to Lo transition
+{
+	return m_cast_ca1;
+}
+
+
+READ_LINE_MEMBER( gi6809_state::cast_pia1_ca2 )  // castaway - coin system (B) - Hi to Lo transition
+{
+	return m_cast_ca2;
+}
+
+
+/*********************************************
+*               Video Hardware               *
+*********************************************/
+
+void gi6809_state::gi6809_videoram_w(offs_t offset, uint8_t data)
+{
+	m_videoram[offset] = data;
+	m_bg_tilemap->mark_tile_dirty(offset);
+}
+
+void gi6809_state::gi6809_colorram_w(offs_t offset, uint8_t data)
+{
+	m_colorram[offset] = data;
+	m_bg_tilemap->mark_tile_dirty(offset);
+}
+
+
+TILE_GET_INFO_MEMBER(gi6809_state::get_bg_tile_info)
+{
+/*  - bits -
+    7654 3210
+    --xx xx--   tiles color.
+    ---- --x-   tiles bank.
+    ---- ---x   tiles extended address (MSB).
+    xx-- ----   unused.
+*/
+
+	int attr = m_colorram[tile_index];
+	int code = ((attr & 1) << 8) | m_videoram[tile_index];
+	int bank = (attr & 0x02) >> 1;      // bit 1 switch the gfx banks
+	int color = (attr & 0x3c) >> 2;     // bits 2-3-4-5 for color
+
+	tileinfo.set(bank, code, color, 0);
+}
+
+void gi6809_state::video_start()
+{
+	m_bg_tilemap = &machine().tilemap().create(*m_gfxdecode, tilemap_get_info_delegate(*this, FUNC(gi6809_state::get_bg_tile_info)), TILEMAP_SCAN_ROWS, 8, 8, 32, 32);
+}
+
+uint32_t gi6809_state::screen_update_gi6809(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect)
+{
+	m_bg_tilemap->draw(screen, bitmap, cliprect, 0, 0);
+	return 0;
+}
+
+
+void gi6809_state::gi6809_palette(palette_device &palette) const
+{
+/*
+    This hardware has a feature called BLUE KILLER.
+    Using the original intensity line, the PCB has a bridge
+    that allows (as default) to turn the background black.
+
+    All games running on this hardware
+    were designed with black background.
+
+    7654 3210
+    ---- ---x   red component.
+    ---- --x-   green component.
+    ---- -x--   blue component.
+    ---- x---   blue killer.
+    xxxx ----   unused.
+*/
+
+	// 0000KBGR
+	uint8_t const *const color_prom = memregion("proms")->base();
+	if (!color_prom)
+		return;
+
+	for (int i = 0; i < palette.entries(); i++)
```

Comment:
The spacing on this `if` and `for` is inconsistent with `if` and `switch` statements elsewhere in the file.